src/Compilers/CSharp/Portable/BoundTree/BoundTreeWalker.cs

```diff
@@ -112,12 +112,10 @@ protected BoundTreeWalkerWithStackGuardWithoutRecursionOnTheLeftOfBinaryOperator
 
             rightOperands.Push(node.Right);
 
-            var binary = (BoundBinaryOperator)node.Left;
-
             BeforeVisitingSkippedBoundBinaryOperatorChildren(binary);
             rightOperands.Push(binary.Right);
 
-            BoundExpression current = binary.Left;
+            BoundExpression? current = binary.Left;
```

Comment:
>? [](http://example.com/codeflow?start=27&length=1)

Is `?` necessary?

Reply:
Yes, `TryPop` will assign `null` when it returns false.